core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsRaftManager.java

```diff
@@ -55,7 +62,7 @@ class JGroupsRaftManager implements RaftManager {
          throw log.forkProtocolRequired();
       }
       this.mainChannel = mainChannel;
-      raftMembers = globalConfiguration.transport().raftMembers();
+      raftMembers = new HashSet<>(globalConfiguration.transport().raftMembers());
```

Comment:
Here is where things get complicated, mainly in k8s. If we deploy a 3 pod cluster, we get members `[A,B,C]` and, when we increase the number of replicas, the operator would invoke "raft add member" with `D` but the 4th pod would start with the original configuration stored in the config map.

Note, it is a theoretical scenario since we don't support RAFT yet. I won't block this PR because of it but we may have to think harder about it. 

My other concern is that `raftMembers` is only updated on the node that invokes `addMember` while the remaining cluster keeps the initial list :)

Reply:
Yeah, I had an idea for the membership a while, but it involved changes for jgroups-raft instead. I thought about adding a learner/witness role. When a node starts and is not in the initial member list, it takes the role of learner. Once all the entries are synchronized and the node receives the membership operation adding itself, it becomes a follower. The learner wouldn't change quorum size and wouldn't count towards replication. It would just receive the entries after the followers commit. But well, quite some changes :/

With the current state, I need to think a little about how we could handle this scenario. Membership changes always have it's corner cases.


> My other concern is that `raftMembers` is only updated on the node that invokes `addMember` while the remaining cluster keeps the initial list :)

I worried about it too, but the operation is replicated in the other state machines. I'll add your suggestion to the `raftMembers()` method, so it should be fine. Before opening the PR, I created an internal state machine of "raft metadata" to store this information. We would apply the operation first on the metadata with RAFT, and then on the user's state machine. This internal state machine would include all the members. But I gave up on the idea at the end :sweat_smile: